server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -275,6 +281,11 @@ public final class NodeChannels extends CloseableConnection {
 
         @Override
         public Version getVersion() {
+            return node.getVersion();
```

Comment:
This changes the meaning of this method. In some cases, the node version does not match the version in the constructor, causing some tests to fail. Is this valid, or should the node version always match the transport version here?

Reply:
This causes some security tests to fail; it may be that when security is updated this problem goes away